designs/2022-docs-information-architecture-update/README.md

```diff
@@ -0,0 +1,352 @@
+- Repo:  eslint/eslint
+- Start Date: 2022-10-19
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Ben Perlmutter (@bpmutter)
+
+# Docs Information Architecture Update
+
+## Summary
+
+This document contains proposed changes to the information architecture (IA) in the ESLint documentation website, <https://eslint.org/docs>.
+
+On a high level, the proposed IA breaks the content into the following sections:
+
+- Use ESLint in Your Project: A refactor of the current [User Guide](https://eslint.org/docs/latest/user-guide/)
+- Extend ESLint: A refactor of the current [Developer Guide](https://eslint.org/docs/latest/developer-guide/), with some content moved to the section Maintain ESLint and Community Contributions
+- Maintain ESLint: Expansion of current [Maintainer Guide](https://eslint.org/docs/latest/maintainer-guide/), with some content taken from the current ‘Developer Guide’.
+- Community Contributions: Expansion of the current [Contributing section](https://eslint.org/docs/latest/developer-guide/contributing/) of the Developer Guide.
+
+## Motivation
+
+The motivation of this proposed change to the ESLint documentation IA is to reorient it around the core personas of the ESLint community.
+
+These personas are:
+
+- **The User**: Someone who wants to use ESLint as it currently exists, including plugins. The proposed 'Use ESLint in Your Project' section addresses this persona's needs.
+- **The Extender**: Someone who wants to extend the functionality of ESLint by creating a plugin, custom formatter, custom parser, sharable configuration, etc. The proposed 'Extend ESLint' section addresses this persona's needs.
+- **The Maintainer**: Someone who wants to contribute to the core ESLint project. The proposed 'Maintain ESLint' section addresses this persona's needs.
```

Comment:
Per my previous comment:

**The Integrator**: Someone who wants to embed ESLint in another application.

Reply:
so for me, the Node.js API makes sense for 'the integrator' persona. though i imagine "extenders" would also use it when testing their extensions.  

however, i see the [integrations page](https://eslint.org/docs/latest/user-guide/integrations) as being more oriented for end users (the user persona). 

per these points, i think it makes sense to keep integrations as they are on this outline. 

though of course, open to have my mind changed on this 😄

Reply:
thinking a little more on this, rather than breaking of "the integrator" as a separate persona, perhaps we can consider the integrator as a "sub-persona" of the "extender", in the same way that a plugin developer and a custom rule developer are "sub-personas" of the extender.

per this, it could make sense to add a 'Create Integrations' page within the 'Extend ESLint' section. of course, adding this page wouldn't be part of the IA project, but could be a good inclusion in Phase 4: Extend ESLint documentation update.

@nzakas what do you think about this option?

Reply:
I just don't agree that integrator is a sub-persona of extender. To me, these are two very different use cases with the only commonality being that they both write JavaScript code to work with ESLint.

* Extender - Uses the core of ESLint and creates new rules, parsers, etc. that ESLint consumes during its normal course of execution. The goal is to extend ESLint with more functionality.
* Integrator - Uses ESLint to extend some other program, such as an IDE, CI system, or build system. The goal here is not to add more functionality to ESLint, but rather to include ESLint in some other program.

I'm not seeing any overlap between these two use cases. Is there some overlap that I'm missing?

Reply:
Oh, missed the last point: totally fine with keeping "Integrations" where they are in the outline. I just feel like we need another section to help people embed ESLint elsewhere.

Reply:
@nzakas thank you for clarifying there. i see what you're saying. makes sense to me as well.

let's add an "Integrate ESLint" section to the docs.

with that being said, i'd rather not include it in this first pass of information architecture updates b/c as it stands right now, the section would just contain the "Node.js API". i'd like to avoid net-new content as part of the information architecture updates, to keep this phase of the project limited in scope. 

per this, i've made a new phase (6) of the [documentation update project](https://github.com/eslint/eslint/issues/16365) to cover the integrator persona. and for the time being, i'd like to keep the Node.js API reference in the "Extend ESLint" section.

if you're good with this approach, i'll also update the RFC document to state that future documentation updates will add "integrator" persona docs that'll include the Node.js API reference.

Reply:
Yup, that makes total sense to me. Thanks!

Reply:
added line about it in this commit - https://github.com/eslint/rfcs/pull/97/commits/dffd504d7608517334d8b60c6c2a4e67d7303374 

and have added phase to document integrator persona in the [documentation update project](https://github.com/eslint/eslint/issues/16365).

resolving this conversation since it seems we're aligned on the next steps here @nzakas.